src/crimson/os/seastore/lba_manager/btree/lba_btree_node_impl.cc

```diff
@@ -404,15 +404,12 @@ LBALeafNode::mutate_mapping_ret LBALeafNode::mutate_mapping(
   ceph_assert(!at_min_capacity());
   auto mutation_pt = find(laddr);
   if (mutation_pt == end()) {
-    ceph_assert(0 == "should be impossible");
-    return mutate_mapping_ret(
-      mutate_mapping_ertr::ready_future_marker{},
-      std::nullopt);
+    return crimson::ct_error::enoent::make();
```

Comment:
Most likely, it means that the next layer up can do something constructive (like printing a warning an cancelling the transaction) instead of crashing.  Most asserts should eventually become useful errors.

Reply:
@athanatos i missed this comment. i mean, if this is a logical error. we should probably should always `assert()`.